### ISSUE TITLE
Support for docker image on arm

### ISSUE THREAD

**Issue by gitrc** [splunk/fluent-plugin-kubernetes-metrics#49]
**What would you like to be added**:
This project does not provide arm images via docker hub.
**Why is this needed**:
I have arm based nodes in my deployment for which I would like to collect logs from.

Thank you!

**Comment by chaitanyaphalak** [comment]
@gitrc Do you need officially supported Splunk ARM docker images for fluent-plugin-kubernetes-metrics. Currently we have https://hub.docker.com/r/splunk/k8s-metrics/tags which is based off alpine.

**Comment by gitrc** [comment]
Hi @chaitanyaphalak I went ahead and published them myself for linux/amd64 and linux/arm64 using docker buildx.

https://hub.docker.com/u/gitrc

It would still be good if Splunk officially supported arm64.

**Comment by chaitanyaphalak** [comment]
I would encourage you to contribute to the project with your DockerFile so that we could include ARM in future releases.

**Comment by gitrc** [comment]
Hi @chaitanyaphalak this is more about implementing https://engineering.docker.com/2019/04/multi-arch-images/ on your side as part of whatever CI/CD process that is shipping the current containers to dockerhub.  I took liberties with regards to the Dockerfile changes in the 3 separate projects (fluent-plugin-splunk-hec, fluent-plugin-kubernetes-metrics, and fluent-plugin-k8s-metrics-agg) just to get this working.  The existing Dockerfiles for these projects build the ruby gems on the host and then copy them into the container.  I modified this to build gems inside the container and switched to using apt upstream for dumb-init instead of pulling precompiled binaries from github.  I do hope that Splunk will support arm64 in the future as part of this product roadmap.  There is definitely a push towards mainstream ARM adoption, with Amazon EC2 A1 instances and the Docker/ARM partnership.  Thank you.
